File: Medium/reverse_integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        strX = str(x)
        isNegative = strX[0] == "-"
        
        if isNegative:
            strX = "-"+strX[:0:-1]
        else:
            strX = strX[::-1]
        
        convToInteger = int(strX)
        
        if convToInteger < -(2**31) or convToInteger > (2**31)-1:
            return 0
        
        return convToInteger

print(Solution().reverse(123)); #321
print(Solution().reverse(-123)); #-321
print(Solution().reverse(1534236469)); #0

File: Medium/reverse_integer.php
<?php

class Solution {

    /**
     * @param Integer $x
     * @return Integer
     */
    function reverse($x) {
        $strX = strval($x);
        $isNegative = $strX[0] == "-";

        if($isNegative) {
            $strX = "-".strrev(substr($strX, 1, strlen($strX)));
        } else {
            $strX = strrev($strX);
        }

        $convToInteger = intval($strX);

        if($convToInteger < -(2**31) || $convToInteger > (2**31) - 1) {
            return 0;
        }

        return $convToInteger;
    }
}

$solution = new Solution;

print_r($solution->reverse(123)."\n"); //321
print_r($solution->reverse(-123)."\n"); //-321
print_r($solution->reverse(1534236469)."\n"); //0